web/modules/custom/weather_block/src/Services/UserWeatherHandler.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+namespace Drupal\weather_block\Services;
+
+use Drupal\Core\Database\Connection;
+use Drupal\Core\Http\ClientFactory;
+use Drupal\Core\Logger\LoggerChannelFactoryInterface;
+
+/**
+ * Save cities for users.
+ */
+class UserWeatherHandler {
+
+  /**
+   * Initialize a database property.
+   */
+  public function __construct(protected Connection $database, protected LoggerChannelFactoryInterface $logger, protected ClientFactory $httpClient) {
+  }
+
+  /**
+   * Saves selected cities for a user in the database.
+   *
+   * @throws \Exception
+   */
+  public function saveCityForUser($userId, $city): void {
+    $this->database->delete('weather_block_city_users')
+      ->condition('user_id', $userId)
+      ->execute();
+
+    $this->database->insert('weather_block_city_users')
+      ->fields([
+        'user_id' => $userId,
+        'city_name' => $city,
+      ])
+      ->execute();
+  }
+
+  /**
+   * Set a default value of city.
+   */
+  public function saveDefaultCity($city) {
+    $this->database->delete('weather_block_defaults')
+      ->condition('default_cit', $city)
```

Comment:
you never deleting previous row, as there are no `default_cit` column
I'm not sure even if this code works without throwing exceptions.
Please test your changes before commiting

Reply:
Still typo, still most likely doesn't work